controller_manager:
  ros__parameters:
    update_rate: 10 # Hz
    use_sim_time: false

    four_wheel_drive_controller:
      type: shelfbot/FourWheelDriveController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

four_wheel_drive_controller:
  ros__parameters:
    # Kinematic parameters
    joint_names:
      - joint_base_axis_front_left
      - joint_base_axis_back_left
      - joint_base_axis_back_right
      - joint_base_axis_front_right
    
    front_left_joint_names: ["joint_base_axis_front_left"]
    back_left_joint_names: ["joint_base_axis_back_left"]
    front_right_joint_names: ["joint_base_axis_front_right"]
    back_right_joint_names: ["joint_base_axis_back_right"]

    wheel_separation: 0.45
    wheel_radius: 0.075

    # Odometry settings
    odom_frame_id: odom
    base_frame_id: base_footprint
    enable_odom_tf: true

    # Command topic and timeout
    cmd_vel_timeout: 0.5 # seconds

    # Velocity and acceleration limits
    linear.x.has_velocity_limits: true
    linear.x.has_acceleration_limits: true
    linear.x.max_velocity: 0.2 # m/s
    linear.x.min_velocity: -0.2 # m/s
    linear.x.max_acceleration: 0.5 # m/s^2
    linear.x.min_acceleration: -0.5 # m/s^2

    angular.z.has_velocity_limits: true
    angular.z.has_acceleration_limits: true
    angular.z.max_velocity: 0.5 # rad/s
    angular.z.min_velocity: -0.5 # rad/s
    angular.z.max_acceleration: 0.5 # rad/s^2
    angular.z.min_acceleration: -0.5 # rad/s^2